core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -260,13 +267,18 @@ class DAGSchedulerSuite
 
   test("zero split job") {
     var numResults = 0
+    var failureReason: Option[Exception] = None
     val fakeListener = new JobListener() {
-      override def taskSucceeded(partition: Int, value: Any) = numResults += 1
-      override def jobFailed(exception: Exception) = throw exception
+      override def taskSucceeded(partition: Int, value: Any): Unit = numResults += 1
+      override def jobFailed(exception: Exception): Unit = {
+        failureReason = Some(exception)
+      }
     }
     val jobId = submit(new MyRDD(sc, 0, Nil), Array(), listener = fakeListener)
     assert(numResults === 0)
     cancel(jobId)
+    assert(failureReason.isDefined)
+    assert(failureReason.get.getMessage() === "Job 0 cancelled ")
```

Comment:
this test used to just log an exception on `cancel(jobId)`.  I'm not sure what it was supposed to be testing before.  I made the minimal change here, by capturing the exception and checking it.  But maybe `cancel(jobId)` should *not* be creating an exception?  Is the idea that if you submit a job with no partitions, it will immediately stop?  That way, if you try to cancel it, you'd just hit [this case](https://github.com/apache/spark/blob/bb1640529725c6c38103b95af004f8bd90eeee5c/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L1224) with a harmless `logDebug`?  That suggests we should change [`handleJobSubmitted`](https://github.com/apache/spark/blob/bb1640529725c6c38103b95af004f8bd90eeee5c/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L757) to handle empty jobs, the same way we handle it in [`submitMissingTasks`](https://github.com/apache/spark/blob/bb1640529725c6c38103b95af004f8bd90eeee5c/core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala#L910) for stages with no partitions.